src/API/Google/Settings.php

```diff
@@ -387,6 +387,53 @@ public function get_store_address(): AccountAddress {
 		return $mc_address;
 	}
 
+	/**
+	 * Check whether the address has errors
+	 *
+	 * @param AccountAddress $address to be validated.
+	 *
+	 * @return array
+	 */
+	public function wc_address_errors( AccountAddress $address ): array {
+		/** @var WC $wc */
+		$wc = $this->container->get( WC::class );
+
+		$countries = $wc->get_wc_countries();
+
+		$locale          = $countries->get_country_locale();
+		$locale_settings = $locale[ $address->getCountry() ] ?? [];
+
+		$fields_to_validate = [
+			'address_1' => $address->getStreetAddress(),
+			'city'      => $address->getLocality(),
+			'country'   => $address->getCountry(),
+			'postcode'  => $address->getPostalCode(),
+			'state'     => $address->getRegion(),
+		];
+
+		return $this->validate_address( $fields_to_validate, $locale_settings );
+	}
+
+	/**
+	 * Check whether the required address fields are empty
+	 *
+	 * @param array $address_fields to be validated.
+	 * @param array $locale_settings locale settings
+	 * @return array
+	 */
+	public function validate_address( array $address_fields, array $locale_settings ): array {
+		$errors = array_filter(
+			$address_fields,
+			function ( $field ) use ( $locale_settings, $address_fields ) {
+				$is_required = $locale_settings[ $field ]['required'] ?? true;
+				return $is_required && empty( $address_fields[ $field ] );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		return array_keys( $errors );
+	}
```

Comment:
[The default **state** of locale settings is required](https://github.com/woocommerce/woocommerce/blob/6.0.0/plugins/woocommerce/includes/class-wc-countries.php#L738-L746), but this does not seem to be correct for all countries. Countries like Lithuania (LT) or Taiwan (TW), which have no **state** in WC Settings and also have no specified locale settings, are impossible to proceed with submission. Because the postcode is always `NULL`.

![image](https://user-images.githubusercontent.com/17420811/148372756-e07230dc-3c3b-47b5-9007-3178e2c4d7b5.png)

Probably the problem belongs to WC core, but it might need to add some additional processing in GLA in advance.

Reply:
As we can not rely on the WC core to check if the state is required or not because WC itself is not setting this field.
I propose to remove the **state** validation so countries like Lithuania or Taiwan would not have this issue. Let me know what you think about this.

Reply:
> I propose to remove the **state** validation so countries like Lithuania or Taiwan would not have this issue. Let me know what you think about this.

I think it will work well. If a country has states, the state will be selected together when choosing a country in WC Settings. So it should be ok to skip the check of **state**.

![image](https://user-images.githubusercontent.com/17420811/148488425-70e4c83b-3f66-4069-a311-f60be728327a.png)

Reply:
I will create a small PR removing the state validation and I will keep the validation of the rest of the fields. Let me know if you agree with this so I can proceed. Thanks @eason9487!

Reply:
Sounds great! Thank you @jorgemd24!